internal/reflect.go

```diff
@@ -31,34 +31,44 @@ func init() {
 	interfaceTyp = reflect.TypeOf(&x).Elem()
 }
 
-// ToArray converts a []interface{} slice into an equivalent fixed-length array
+// MakeHashable converts a []interface{} slice into an equivalent fixed-length array
 // [...]interface{} for use as a comparable map key
-//
-func ToArray(s []interface{}) interface{} {
-	switch len(s) {
+func MakeHashable(s []interface{}) interface{} {
+	d := make([]interface{}, len(s))
+	// Convert byte slices into strings as they are otherwise not comparable/hashable.
+	for i, elem := range s {
+		if b, ok := elem.([]byte); ok {
+			d[i] = string(b)
+		} else {
+			d[i] = elem
+		}
+	}
+
+	// Return arrays as they are comparable/hashable.
+	switch len(d) {
 	// fast code paths for short arrays:
 	case 0:
 		return [...]interface{}{}
 	case 1:
-		return [...]interface{}{s[0]}
+		return [...]interface{}{d[0]}
```

Comment:
super nit: consider changing out `s` -> `d` in the arguments instead so that these references stay the same.

Reply:
(i also checked _find all references_ so it seems fine to leave)